src/types/wantlist/entry.js

```diff
@@ -41,15 +34,12 @@ class WantListEntry {
   }
 
   /**
-   * @param {API} other
-   * @returns {boolean}
+   * @param {any} other
```

Comment:
```suggestion
   * @param {unknown} other
```

We are using unknown for such scenarios in libp2p. Perhaps better?

Reply:
I don't think you can access properties of a variable of type `unknown`?

Reply:
We do it here https://github.com/libp2p/js-libp2p/blob/master/src/record/peer-record/index.js#L68
does not seem problematic